### PR TITLE
Add automated error capturing with video

### LINKED ISSUE #618: [FEATURE]: Automated Error Capturing with Video
### Feature summary

Automated Error Capturing with Video

### Feature description

As of now we need to find error log traces from logs, we need to be activity monitor screen for taking screenshots.

proposal is to have error capturing in automated way. whenever code hits error It should create a folder for that error having log file while contains error + x (may be 2000) line above error for debugging purposes. Along with that a video of browsers what happening on the screen.

Creating error video can be done easily using selenium (as this is widely used for end to end testing).

### Motivation

_No response_

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yusefes**
Fixes #618

Add automated error capturing with video recording.

* Add `opencv-python` to `requirements.txt` for video recording.
* Add `capture_error` function in `src/utils.py` to handle error capturing and video recording.
* Update `main.py` to call `capture_error` in exception handling blocks.
* Update `src/aihawk_easy_applier.py` to call `capture_error` in exception handling blocks.
* Update `src/aihawk_job_manager.py` to call `capture_error` in exception handling blocks.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/feder-cr/Auto_Jobs_Applier_AIHawk/issues/618?shareId=f2ee2ede-9d3e-4f9f-842a-03c6c96da9f3).

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -166,3 +168,33 @@ def printyellow(text):
...
+    # Save the error message to a log file
+    log_file_path = os.path.join(error_dir, "error.log")
+    with open(log_file_path, "w") as log_file:
+        log_file.write(error_message)
+
+    # Capture a screenshot
+    screenshot_path = os.path.join(error_dir, "screenshot.png")
+    driver.save_screenshot(screenshot_path)
+
+    # Record a video of the browser screen
```
Comment (surapuramakhil) [inline src/utils.py:187]: move line 187 to 196 to separate method - call it as record browser screen, you  can have time as param (up to you)

File: src/utils.py [hunk 2]
```diff
@@ -166,3 +168,33 @@ def printyellow(text):
...
+    error_dir = os.path.join("error_logs", timestamp)
+    os.makedirs(error_dir, exist_ok=True)
+
+    # Save the error message to a log file
+    log_file_path = os.path.join(error_dir, "error.log")
+    with open(log_file_path, "w") as log_file:
+        log_file.write(error_message)
+
+    # Capture a screenshot
+    screenshot_path = os.path.join(error_dir, "screenshot.png")
```
Comment (surapuramakhil) [inline src/utils.py:184]: move this to separate method - capture_browser_screenhot

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
if possible, can you create screen_record_utils.py and move screen recording code over there.

Also create error_utils.py

**Comment by surapuramakhil** [comment]
@yusefes great work.

**Comment by surapuramakhil** [comment]
@yusefes also this PR has failed in some testcases - can you also fix text cases before you push your changes.

**Comment by surapuramakhil** [comment]
@yusefes very unfortunate you closed this PR.

**Comment by feder-cr** [comment]
@yusefes Hi, do you want to work on this?

For better communication with other open source developers I suggest you join our discord group: https://discord.gg/mMZcMTH9K6

**Pull request closed without merging**